promql/query_logger.go

```diff
@@ -83,7 +83,14 @@ func logUnfinishedQueries(filename string, filesize int, logger log.Logger) {
 func getMMapedFile(filename string, filesize int, logger log.Logger) ([]byte, error) {
 	file, err := os.OpenFile(filename, os.O_CREATE|os.O_RDWR|os.O_TRUNC, 0o666)
 	if err != nil {
-		level.Error(logger).Log("msg", "Error opening query log file", "file", filename, "err", err)
+		// Try and log a full path to aid in debugging, even if we received a
+		// relative one. if filepath.Abs returns an error, we just log the base
+		// path as received.
+		fullpath, err := filepath.Abs(filename)
+		if err == nil {
+			fullpath = filename
+		}
+		level.Error(logger).Log("msg", "Error opening query log file", "file", filename, "fullpath", fullpath, "err", err)
```

Comment:
```suggestion
		level.Error(logger).Log("msg", "Error opening query log file", "file", absPath, "err", err)
```

Reply:
I wasn't sure about omitting the path-as-given, the user might think "but that's not what I specified". But I'll defer to you on that.

Reply:
It is still in the error message: --config.file=%s, so that's good.